### DISCUSSION TITLE
Milvus 2.0 Depolyment Problem: RootCoord Can't start

### DISCUSSION THREAD

**Discussion by orange-jams** [milvus-io/milvus#6318] [Q&A and General discussion]
Due to some reaseon, we can't use helm to deploy distributed milvus2.0.So we follow helm's yaml files to deploy each component.When deploying RootCoord error occurs, the error message is "function CompareAndSwap error for compare is false for key: RootCoord".

**Comment by xiaofan-luan** [comment]
@neza2017 any ideas?

**Comment by neza2017** [comment]
for `milvus 2.0` every component will register itself into etcd with lease, and etcd will remove the registered information automatically  after component disconnect with etcd longer than 10s
so this error message means that the someone has registered `RootCoord` in th etcd, so you failed to deploy the `RootCoord`
may be you deploy `RootCoord` more than once, and the second `RoocCoord` reports this error.

for the default config, the `key` for `RootCoord` in etcd is `by-dev/meta/session/RootCoord`, please check it this key exist in etcd.

**Reply by orange-jams** [reply]
Thanks for your reply, we check the files in etcd,but not found `by-dev/meta/session/RootCoord`. Maybe etcd we deployed is false, could you provide etcd's `deplyment.yaml` which we can't find from milvs-helm.

**Reply by jeffoverflow** [reply]
@orange-jams 
helm generated yamls are good sources of examples. use these command:

```shell
git clone https://github.com/milvus-io/milvus-helm.git
# Helm >= v3.x
helm template my-release ./milvus-helm/charts/milvus > ./milvus_generated.yaml
# edit milvus_generated.yaml if necessary
```
The `milvus_generated.yaml` contains the etcd's `deplyment.yaml`

**Reply by orange-jams** [reply]
As i said in description due to some reaseon, we can't use helm to deploy distributed milvus2.0. So what you suggest, we can't have a try.

**Reply by jeffoverflow** [reply]
@orange-jams 
[milvus_generated.yaml.txt](https://github.com/milvus-io/milvus/files/6801183/milvus_generated.yaml.txt)
